avm/res/network/network-watcher/connection-monitor/main.bicep

```diff
@@ -0,0 +1,86 @@
+metadata name = 'Network Watchers Connection Monitors'
+metadata description = 'This module deploys a Network Watcher Connection Monitor.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Optional. Name of the network watcher resource. Must be in the resource group where the Flow log will be created and same region as the NSG.')
+param networkWatcherName string = 'NetworkWatcher_${resourceGroup().location}'
+
+@description('Required. Name of the resource.')
+param name string
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. List of connection monitor endpoints.')
+param endpoints array = []
+
+@description('Optional. List of connection monitor test configurations.')
+param testConfigurations array = []
+
+@description('Optional. List of connection monitor test groups.')
+param testGroups array = []
+
+@description('Optional. Specify the Log Analytics Workspace Resource ID.')
+param workspaceResourceId string = ''
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableTelemetry bool = true
+
+var outputs = !empty(workspaceResourceId) ? [
```

Comment:
It appears this var is only used in a single location. Might make sense to move it's value to where it is used. Makes it less fragmented ;)